add volume backingstore testcases

MalloZup opened the pull request:
improve codecoverage for backingstore volume

dmacvicar commented:
What are you exactly testing here?

I think the test should then retrieve the volume definition from _libvirt_ and check that the base volume was in fact used as a `backingStore` element..

MalloZup commented:
@dmacvicar ok thx for hint

MalloZup commented:
@dmacvicar  thank you for your remark.

I addressed your comment and we check in the libvirtxml definition if the volume is a backingstore now.

MalloZup commented on libvirt/resource_libvirt_volume_test.go:
```diff
@@ -82,6 +83,46 @@ func testAccCheckLibvirtVolumeDoesNotExists(n string, volume *libvirt.StorageVol
+		if err != nil {
+			return fmt.Errorf("Error reading libvirt volume XML description: %s", err)
+		}
+		if volumeDef.BackingStore == nil {
```
actual test for backing store is here

MalloZup commented:
@dmacvicar feel free to review it. Now make more sense, we are checking the backinstore xml 

TIA :sun_with_face: :lemon: :seedling: :tropical_fish:

MalloZup commented:
@dmacvicar  i am merging this in hope to make our review lightweight especially for testacc. 
I mean since this is not a huge major new-feature, is ok that i don"t wait to much for it, since it can improve our testcoverage/stability of provider.

On the testcase, i adressed your review/comment, with https://github.com/dmacvicar/terraform-provider-libvirt/pull/363/files/5b8287c04fd2c8bf2053174d1963fe9afe44ac45#r213265837, which we are really testing now the xml defintion.
